### PR TITLE
getRUN contract, bootstrap integration (WIP)

### PR DESCRIPTION
**Pull request by dckc**
refs #3788

cc @Chris-Hibbert 

## Description

related issues:

- [ ] new issue? no calls to `sortedVaultKits.remove`. every vault is charged interested at every epoch - is that sufficiently scalable?
- [ ] ISSUE: why aren't HIGH_FEE etc. exported from @agoric/zoe?
- [ ] `@ts-ignore` governance wrapper obscures publicFace type

### Security Considerations

 - has access to RUN mint

### Testing Considerations

 - [ ] accrue interest
 - [ ] golang bridge for lien
 - [ ] partial payoff

### CODE REVIEW

File: packages/treasury/src/runLoC.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * @param {{ feeMintAccess: FeeMintAccess }} privateArgs
+ */
+const start = async (zcf, { feeMintAccess }) => {
+  const { main: initialValue } = zcf.getTerms();
+
+  const {
+    makePublicFacet,
+    makeCreatorFacet,
+    getParamValue,
+  } = handleParamGovernance(zcf, harden(initialValue));
```
Comment (Chris-Hibbert) [inline packages/treasury/src/runLoC.js]: This is taking the creator's word for what the governed parameters should look like. 

The AMM declares its expectations in [params.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/vpool-xyk-amm/params.js), in a way that is used in the contract, and should have been used in the tests when creating  parameters for terms.

[stablecoinMachine](https://github.com/Agoric/agoric-sdk/blob/58c37780a506dfafc5356b3edcc8e13edfc537a6/packages/treasury/src/stablecoinMachine.js#L98) asserts that the governed params from terms match the local declaration to ensure that all and only the required params are present. If there are params that aren't used or verified at startup, this seems crucial to me.

File: packages/treasury/src/runLoC.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  const getRatio = name => {
+    const x = getParamValue(name);
+    assertIsRatio(x);
+    return /** @type { Ratio } */ (x);
+  };
```
Comment (Chris-Hibbert) [inline packages/treasury/src/runLoC.js]: Seems generally useful. I'm fine with generalizing and adding these into the governance pkg.

Is it already time to refactor `paramManager` so it splits out all the different types? Or should we add helper tools like this for a while longer?

**Review by Chris-Hibbert — COMMENTED** [review]
a few thoughts on the use of governance. I'll wait till you're ready before I do a full review

File: packages/treasury/src/runLoC.js (hunk 2 above)
Comment (dckc) [reply packages/treasury/src/runLoC.js]: Now would be a good time for me, while it's swapped in. And before we dig the hole deeper by using this API in more contracts.

File: packages/treasury/src/runLoC.js (hunk 1 above)
Comment (dckc) [reply packages/treasury/src/runLoC.js]: I don't understand how the `sameStructure` in stablecoinMachine works; it seems to be comparing the list `['ChargingPeriod', ...]` with something that looks like `[{name, type, value}, ...]`.

Comment (Chris-Hibbert) [reply packages/treasury/src/runLoC.js]: They should both be `{ "vaultParams": ["ChargingPeriod", "RecordingPeriod", "InitialMargin", "LiquidationMargin", "InterestRate", "LoanFee"] }`, which comes directly from params.js.

When setting up the treasury in test-stablecoin, treasuryTerms contains `governedParams: governedParameterTerms`, where the latter is imported from params.

**Comment by Tartuffo** [comment]
@dckc @Chris-Hibbert   When I first started working on our ZH board, I marked this as being for the Mainnet 1 release because it was already in the Review/QA pipeline. This PR is from Nov. Should this be closed?

**Comment by dckc** [comment]
Closed? Why?

**Pull request closed without merging**